src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -840,13 +840,25 @@ private StringBuffer preMigrationChecks(StringBuffer itemStatus,
 				if (api == null) {
 					// exit if no Box API connection could be made
 					// returning all changed variables
+					itemStatus.append("Cannot create Box folder for folder " + title);
+					
 					HashMap<String, Object> rv = new HashMap<String, Object>();
-					rv.put("itemStatus", "Cannot create Box folder for folder " + title);
+					rv.put("itemStatus", itemStatus);
 					rv.put("containerStack", containerStack);
 					rv.put("boxFolderIdStack", boxFolderIdStack);
 					return rv;
 				}
 
+				if (title == null) {
+					// exit if folder title is null
+					itemStatus.append("Cannot create Box folder for null folder title");
+					HashMap<String, Object> rv = new HashMap<String, Object>();
+					rv.put("itemStatus", itemStatus);
+					rv.put("containerStack", containerStack);
+					rv.put("boxFolderIdStack", boxFolderIdStack);
+					return rv;
```

Comment:
Above map keys can be stored as constants as these are repeated `api=null` check. Also the lines 855-858 can also be extracted to a method can be reused in both if sections